app/connectors/IndividualDetailsConnector.scala

```diff
@@ -19,31 +19,93 @@ package connectors
 import cats.data.EitherT
 import com.google.inject.{Inject, Singleton}
 import config.FrontendAppConfig
+import models.individualDetails.*
 import play.api.Logging
+import repositories.IndividualDetailsRepoTrait
 import uk.gov.hmrc.http.HttpReads.Implicits.*
 import uk.gov.hmrc.http.client.HttpClientV2
 import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, StringContextOps, UpstreamErrorResponse}
 
 import scala.concurrent.{ExecutionContext, Future}
 
+trait IndividualDetailsConnector {
+  def getIndividualDetails(nino: String, resolveMerge: String)(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): EitherT[Future, UpstreamErrorResponse, IndividualDetails]
+}
+
 @Singleton
-class IndividualDetailsConnector @Inject() (
-  val httpClient: HttpClientV2,
+class DefaultIndividualDetailsConnector @Inject() (
+  httpClient: HttpClientV2,
   appConfig: FrontendAppConfig,
   httpClientResponse: HttpClientResponse
-) extends Logging {
+) extends IndividualDetailsConnector
+    with Logging {
 
-  def getIndividualDetails(nino: String, resolveMerge: String)(implicit
+  override def getIndividualDetails(nino: String, resolveMerge: String)(implicit
     hc: HeaderCarrier,
     ec: ExecutionContext
-  ): EitherT[Future, UpstreamErrorResponse, HttpResponse] = {
+  ): EitherT[Future, UpstreamErrorResponse, IndividualDetails] = {
     val url =
       s"${appConfig.individualDetailsServiceUrl}/find-my-nino-add-to-wallet/individuals/details/NINO/${nino.take(8)}/$resolveMerge"
 
-    httpClientResponse.read(
-      httpClient
-        .get(url"$url")
-        .execute[Either[UpstreamErrorResponse, HttpResponse]]
-    )
+    httpClientResponse
+      .read(
+        httpClient
+          .get(url"$url")
+          .execute[Either[UpstreamErrorResponse, HttpResponse]]
+      )
+      .map(_.json.as[IndividualDetails])
+  }
+}
+
+@Singleton
+class CachingIndividualDetailsConnector @Inject() (
+  @javax.inject.Named("default") underlying: IndividualDetailsConnector,
+  repo: IndividualDetailsRepoTrait
+)(implicit ec: ExecutionContext)
+    extends IndividualDetailsConnector
+    with Logging {
+
+  override def getIndividualDetails(nino: String, resolveMerge: String)(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): EitherT[Future, UpstreamErrorResponse, IndividualDetails] =
+    underlying.getIndividualDetails(nino, resolveMerge)
+
+  def getIndividualDetailsWithCache(nino: String, sessionId: String)(implicit
```

Comment:
This is not in the trait. It should be getIndividualDetails like the implementation in other services.